contrib/epee/include/net/levin_protocol_handler_async.h

```diff
@@ -503,8 +527,13 @@ class async_protocol_handler
             else
               m_config.m_pcommands_handler->notify(m_current_head.m_command, buff_to_invoke, m_connection_context);
           }
+          // reuse small buffer
+          if (!temp.empty() && temp.capacity() <= 64 * 1024)
+          {
+            temp.clear();
+            m_fragment_buffer = std::move(temp);
+          }
```

Comment:
It is up to the implementation, but every implementation implements the `std::vector` requirements (never change capacity). The reason is _probably_ to maintain predictable behavior. Always freeing is less performant in many use cases, and sometimes freeing makes it harder for the programmer to optimize around `reserve`/`clear`, etc.

It might be worth dropping since its not **actually** guaranteed. The code is also minimal at least, and all existing implementations currently re-use the memory.